src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlConnection.cs

```diff
@@ -126,11 +126,23 @@ public static SqlConnection Create(IPersistentStorageFaultInjector? faultInjecto
                 // uri of the DB on disk we're associating this in-memory cache with.  This throws on at least OSX for
                 // reasons that aren't fully understood yet.  If more details/fixes emerge in that linked issue, we can
                 // ideally remove this and perform the attachment uniformly on all platforms.
-                var attachString = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-                    ? $"attach database '{new Uri(databasePath).AbsoluteUri}?mode=memory&cache=shared' as {Database.WriteCache.GetName()};"
-                    : $"attach database 'file::memory:?cache=shared' as {Database.WriteCache.GetName()};";
 
-                connection.ExecuteCommand(attachString);
+                // Database name may itself contain portions that could affect the sql statement, so use a prepared
+                // statement here.
```

Comment:
The "here" is troubling.. are there still other places where we don't use a prepared statement? Shouldn't they just all be upgraded just in case?

Reply:
fair question.  let me audit.  we shoudl be using prepared in all cases... but obviously that was not the case.

Reply:
ok.  audited.  all the other places we call ExecuteCommand, it's always effectivelya  constant string.  no variable input flowing in.